Plugins/Public/playercntl_plugin/PimpShip.cpp

```diff
@@ -287,69 +345,220 @@ namespace PimpShip
+					endAt = mapInfo[iClientID].mapCurrEquip.size();
+				}
+				else if (it == firstArg.begin())
+				{
+					beginFrom = 1;
+					endAt = ToUInt(firstArg.substr(index));
 
-		wstring wscCharName = (const wchar_t*) Players.GetActiveCharacterName(iClientID);
+					if (endAt < beginFrom)
+						swap(endAt, beginFrom);
+				}
+				else
+				{
+					beginFrom = ToUInt(firstArg.substr(0, index));
+					endAt = ToUInt(firstArg.substr(index));
 
-		// Check the that player is in a ship dealer.
-		if (!mapInfo[iClientID].bInPimpDealer)
-			return false;
+					if (endAt < beginFrom)
+						swap(endAt, beginFrom);
+				}
 
-		// Charge for the equipment pimp.
-		if (set_iCost > 0)
-		{
-			int iCash = 0;
-			if ((err = HkGetCash(wscCharName, iCash)) != HKE_OK)
+				break;
+			}
+			else if (*it == '*')
+			{
+				if (beginFrom == 0)
+				{
+					endAt = mapInfo[iClientID].mapCurrEquip.size();
+
+					if (it == firstArg.begin())
+					{
+						beginFrom = 1;
+						everyN = ToUInt(firstArg.substr(index));
+					}
+					else if (it == firstArg.end() - 1)
+					{
+						beginFrom = 1;
+						everyN = ToUInt(firstArg.substr(0, index));
+					}
+					else
+					{
+						beginFrom = ToUInt(firstArg.substr(0, index));
+						everyN = ToUInt(firstArg.substr(index));
+
+						if (beginFrom == 0)
+							break;
+					}
+				}
+				else
+				{
+					if (it == firstArg.end() - 1)
+						endAt = mapInfo[iClientID].mapCurrEquip.size();
+					else
+					{
+						endAt = ToUInt(firstArg.substr(index));
+
+						if (endAt < beginFrom)
+							swap(endAt, beginFrom);
+					}
+
+					break;
+				}
+			}
+			else
 			{
-				PrintUserCmdText(iClientID, L"ERR %s", HkErrGetText(err).c_str());
+				PrintUserCmdText(iClientID, L"ERR Invalid syntax");
 				return true;
 			}
-			if (iCash<0 && iCash<set_iCost)
+		}
+
+		if (everyN == 0)
+		{
+			PrintUserCmdText(iClientID, L"ERR Zero advancement");
+			return true;
+		}
+
+		if ((beginFrom == 0) && endAt == 0)
+		{
+			beginFrom = endAt = ToUInt(firstArg);
+			if (beginFrom == 0 || beginFrom > mapInfo[iClientID].mapCurrEquip.size())
 			{
-				PrintUserCmdText(iClientID, L"ERR Insufficient credits");
+				PrintUserCmdText(iClientID, L"ERR hardpoint index is out of bounds");
 				return true;
 			}
-			HkAddCash(wscCharName, 0-set_iCost);
 		}
 
-		// Remove all lights.
-		for (map<uint, EQ_HARDPOINT>::iterator i = mapInfo[iClientID].mapCurrEquip.begin();
-			i != mapInfo[iClientID].mapCurrEquip.end(); ++i)
+		if (beginFrom == 0 || beginFrom > mapInfo[iClientID].mapCurrEquip.size())
 		{
-			pub::Player::RemoveCargo(iClientID, i->second.sID, 1);
+			PrintUserCmdText(iClientID, L"ERR Beginning is out of bounds");
+			PrintUserCmdText(iClientID, L"You may want to use following syntax to select all hardpoints from beginning to %u:", endAt);
+			if (everyN == 1)
+				PrintUserCmdText(iClientID, L"/setitem -%u %u", endAt, iSelectedItemID);
+			else
+				PrintUserCmdText(iClientID, L"/setitem *%u*%u %u", everyN, endAt, iSelectedItemID);
```

Comment:
```suggestion
				PrintUserCmdText(iClientID, L"/setitem -%u*%u %u", endAt, everyN, iSelectedItemID);
```
?

Reply:
No, I am wrong. We should talk about whether the syntax is too complicated though.